src/mono/mono/mini/interp/interp.c

```diff
@@ -3998,7 +4003,7 @@ mono_interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClause
 					MonoObject *this_arg = del->target;
 
 					// replace the MonoDelegate* on the stack with 'this' pointer
-					if (m_class_is_valuetype (this_arg->vtable->klass) && m_class_is_valuetype (cmethod->method->klass)) {
+					if (m_class_is_valuetype (cmethod->method->klass)) {
```

Comment:
The first check was not needed ?

Reply:
To me it seems redundant. I can't think why it would be needed. If you can think on why it would be wrong let me know. I don't plan to backport this part of the change though.